packages/internal/src/js-utils.js

```diff
@@ -87,3 +91,27 @@ export const makeMeasureSeconds = currentTimeMillisec => {
   };
   return measureSeconds;
 };
+
+/**
+ * @param {CopyRecord | Primitive} obj
+ * @returns {string}
+ */
+export const stringifyWithBigint = obj =>
+  JSON.stringify(obj, (key, value) => {
+    return typeof value === 'bigint' ? value.toString() : value;
+  });
+
+/** @type {(str: string) => boolean} */
+const isOnlyNumbers = str => /^\d+$/.test(str);
+
+/**
+ * @param {string} str
+ * @returns unknown
+ */
+export const parseWithBigint = str =>
```

Comment:
I don't like that this approach is vulnerable to collisions and "upgrades" _any_ all-decimal-digits string to a bigint, even if it was user input that should remain a string. What's the reason for not using smallcaps?
```js
const rejectOCap = cap => Fail`${cap} is not pure data`;
const { toCapData, fromCapData } =
  makeMarshal(rejectOCap, rejectOCap, { serializeBodyFormat: "smallcaps" });
const serialize = value => toCapData(value).body.slice(1);
const unserialize = text => fromCapData({ body: `#${text}`, slots: [] });

const value = harden({ number: 42, bigint: 42n, strings: ["42", "+42"] });
const serialized = serialize(value);
const restored = unserialize(serialized);
console.log({ value, serialized, restored });
/* =>
{
  value: { number: 42, bigint: 42n, strings: [ '42', '+42' ] },
  serialized: '{"bigint":"+42","number":42,"strings":["42","!+42"]}',
  restored: { bigint: 42n, number: 42, strings: [ '42', '+42' ] }
}
*/
```

Reply:
I think it's a good idea worth exploring more. There are lots of trade-offs to consider so I'm opting to take it out of the critical path for FU by using the more verbose but well trodden "serialized CapData".

Reply:
For ref, I track this possible optimization in/around..

 - #7999